apps/src/javalab/CommitDialog.jsx

```diff
@@ -3,35 +3,50 @@ import PropTypes from 'prop-types';
 import i18n from '@cdo/javalab/locale';
 import color from '@cdo/apps/util/color';
 import {CompileStatus} from './constants';
-import StylizedBaseDialog from '@cdo/apps/componentLibrary/StylizedBaseDialog';
+import StylizedBaseDialog, {
+  FooterButton
+} from '@cdo/apps/componentLibrary/StylizedBaseDialog';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 
 export default class CommitDialog extends React.Component {
   state = {
     filesToCommit: [],
+    filesToBackpack: [],
     commitNotes: null
   };
 
   renderFooter = buttons => {
     let compileStatusContent = '';
-    switch (this.props.compileStatus) {
-      case CompileStatus.LOADING:
-        compileStatusContent = i18n.compiling();
-        break;
-      case CompileStatus.SUCCESS:
-        compileStatusContent = [
-          <FontAwesome
-            key="icon"
-            icon="check-circle"
-            className="fa-2x"
-            style={styles.iconSuccess}
-          />,
-          <span key="text">{i18n.allFilesCompile()}</span>
-        ];
-        break;
-      case CompileStatus.ERROR:
-        compileStatusContent = i18n.compileFailed();
-        break;
+    let commitText = 'Commit';
+    let isDisabled = true;
+    let showCompileStatus = false;
+    if (this.state.commitNotes) {
+      isDisabled = false;
+    }
+    if (this.state.filesToBackpack.length > 0) {
+      commitText = 'Commit & Save';
+      showCompileStatus = true;
```

Comment:
I think we should only compile once the user clicks 'commit and save'. We don't want to send a bunch of separate compile requests each time they check/uncheck a file, especially because the compilation may not work until they choose a certain combination of files (for the case where one class references another).

Reply:
Okay- the current figma design has the commit check with each combination of files, and then the commit & save button saves the files and closes the modal. With this change, when would we show the "all files compile" status?

Reply:
slack thread: https://codedotorg.slack.com/archives/C01EF4GJ9GE/p1627339139228900

Reply:
Ohh yes okay I forgot we were eliminating that view. Okay so I'll make it so we only see something there if there's a compilation error. So then the commit & save button will not save and close; it will try to save and maybe close!